src/pages/settings/Profile/PersonalDetails/AddressPage.js

```diff
@@ -52,179 +52,159 @@ const defaultProps = {
-                    submitButtonText={this.props.translate('common.save')}
-                    enabledWhenOffline
-                >
-                    <View style={styles.mb4}>
-                        <AddressSearch
-                            inputID="addressLine1"
-                            label={this.props.translate('common.addressLine', {lineNumber: 1})}
-                            defaultValue={street1 || ''}
-                            isLimitedToUSA={false}
-                            renamedInputKeys={{
-                                street: 'addressLine1',
-                                street2: 'addressLine2',
-                                city: 'city',
-                                state: 'state',
-                                zipCode: 'zipPostCode',
-                                country: 'country',
-                            }}
-                            maxInputLength={CONST.FORM_CHARACTER_LIMIT}
-                        />
-                    </View>
-                    <View style={styles.mb4}>
-                        <TextInput
-                            inputID="addressLine2"
-                            label={this.props.translate('common.addressLine', {lineNumber: 2})}
-                            defaultValue={street2 || ''}
-                            maxLength={CONST.FORM_CHARACTER_LIMIT}
+    return errors;
+};
+
+/**
+ * Submit form to update user's first and last legal name
+ * @param {Object} values - form input values
+ */
+const updateAddress = (values) => {
+    PersonalDetails.updateAddress(values.addressLine1.trim(), values.addressLine2.trim(), values.city.trim(), values.state.trim(), values.zipPostCode.trim(), values.country);
+};
+
+const AddressPage = (props) => {
+    const [countryISO, setCountryISO] = useState(PersonalDetails.getCountryISO(lodashGet(props.privatePersonalDetails, 'address.country')) || CONST.COUNTRY.US);
+    const isUsaForm = countryISO === CONST.COUNTRY.US;
+
+    const zipSampleFormat = lodashGet(CONST.COUNTRY_ZIP_REGEX_DATA, [countryISO, 'samples'], '');
+    const zipFormat = props.translate('common.zipCodeExampleFormat', {zipSampleFormat});
+
+    const address = lodashGet(props.privatePersonalDetails, 'address') || {};
+    const [street1, street2] = (address.street || '').split('\n');
+
+    return (
+        <ScreenWrapper includeSafeAreaPaddingBottom={false}>
+            <HeaderWithCloseButton
+                title={props.translate('privatePersonalDetails.homeAddress')}
+                shouldShowBackButton
+                onBackButtonPress={() => Navigation.navigate(ROUTES.SETTINGS_PERSONAL_DETAILS)}
+                onCloseButtonPress={() => Navigation.dismissModal(true)}
+            />
+            <Form
+                style={[styles.flexGrow1, styles.ph5]}
+                formID={ONYXKEYS.FORMS.HOME_ADDRESS_FORM}
+                validate={(values) => validate(props.translate, isUsaForm, values)}
```

Comment:
This lambda is being re-created on every render. Maybe we can use `useCallback` to create a stable reference and pass it to Form.

Reply:
Yes, I'll change that. thanks!

Reply:
Update!